tensorboard/plugins/scalar/scalars_demo.py

```diff
@@ -92,12 +92,14 @@ def run(logdir, run_name,
              description='The change in temperature from the previous '
                          'step, in Kelvins.')
 
-  # Now, augment the current temperature by this delta that we computed.
-  update_step = temperature.assign_add(delta)
-
   # Collect all the scalars that we want to keep track of.
   summ = tf.summary.merge_all()
 
+  # Now, augment the current temperature by this delta that we computed,
+  # blocking the assignment on summary collection to avoid race conditions.
```

Comment:
Just to hammer the point, maybe clarify here that we use summary values computed before the update occurs.